src/lib/util/simh_tape_file.cpp

```diff
@@ -0,0 +1,624 @@
+// license:BSD-3-Clause
+// copyright-holders:Mietek Bak
+
+// best read together with SIMH magtape spec (rev 17 Jan 2022)
+// http://simh.trailing-edge.com/docs/simh_magtape.pdf
+
+#include <cassert>
+#include <cstring>
+#include <stdexcept>
+#include <errno.h>
+#include <fcntl.h>
```

Comment:
As was mentioned in global comments, please use MAME's file I/O rather than POSIX or other direct platform calls.